clientd3d/music.c

```diff
@@ -145,23 +155,74 @@ DWORD OpenMidiFile(const char *lpszMIDIFileName)
       debug(("Unable to get current directory!\n"));
 
    sprintf(filename, "%s%s", current_dir, lpszMIDIFileName);
-   debug(("music filename = %s\n", filename));
+   debug(("music filename = %s \n", filename));
+   // Is it a background music or gameplay element music file?
+   switch (musicType) {
+      case BACKGROUND_MUSIC:
+         if (playing_music != False) 
+         {
+            mciSendCommand(midi_bg_music_element, MCI_CLOSE, 0, 0);
+         }
+         break;
+      case GAMEPLAY_MUSIC:
+         if (playing_midi != False) 
+         {
+            mciSendCommand(midi_element, MCI_STOP, 0, 0);
+            if (dwReturn = mciSendCommand(midi_element, MCI_CLOSE, 0, 0))
+            {
+               // Just debug don't return
+               debug(("mciClose problem for midi_element\n"));
+            }
+         }
+         break;
+      default:
+         debug(("OpenMidiFile: Invalid musicType\n"));
```

Comment:
It's a good idea to log the value of musicType here to help in debugging.

Reply:
ok thanks - done.